File: backend/services/agent-service/app/schemas/response.py
from __future__ import annotations
from typing import Any
from pydantic import BaseModel


class ApiResponse(BaseModel):
    success:    bool
    message:    str
    statusCode: int
    data:       Any = None

    @classmethod
    def ok(cls, data: Any = None, message: str = "") -> "ApiResponse":
        return cls(success=True, message=message, statusCode=200, data=data)

    @classmethod
    def created(cls, data: Any = None, message: str = "") -> "ApiResponse":
        return cls(success=True, message=message, statusCode=201, data=data)

    @classmethod
    def error(cls, status_code: int, message: str) -> "ApiResponse":
        return cls(success=False, message=message, statusCode=status_code, data=None)

File: backend/services/auth-service/handlers/response.go
package handlers

import (
	"net/http"

	"auth-service/models"

	"github.com/gin-gonic/gin"
)

func respondError(c *gin.Context, status int, message string) {
	c.JSON(status, models.APIResponse{
		Success: false,
		Message: message,
	})
}

func respondSuccess(c *gin.Context, message string, data any) {
	c.JSON(http.StatusOK, models.APIResponse{
		Success: true,
		Message: message,
		Data:    data,
	})
}

// bindJSON binds the request body into dest. On failure it writes a 400 response
// and returns false.
func bindJSON(c *gin.Context, dest any) bool {
	if err := c.ShouldBindJSON(dest); err != nil {
		respondError(c, http.StatusBadRequest, "Invalid request: "+err.Error())
		return false
	}
	return true
}
